tests/js/spec/views/dashboardsV2/view.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import {browserHistory} from 'react-router';
+
+import {initializeOrg} from 'sentry-test/initializeOrg';
+import {mountWithTheme} from 'sentry-test/reactTestingLibrary';
+
+import ViewEditDashboard from 'sentry/views/dashboardsV2/view';
+
+describe('Dashboards > ViewEditDashboard', function () {
+  const initialData = initializeOrg();
+
+  it('removes widget params from url and preserves selection params', async function () {
+    const location = {
+      pathname: '/',
+      query: {
+        environment: 'canary',
+        period: '7d',
+        project: '11111',
+        start: null,
+        end: null,
+        utc: null,
+        displayType: 'line',
+        interval: '5m',
+        queryConditions: '',
+        queryFields: 'count()',
+        queryNames: '',
+        queryOrderby: '',
+        title: 'test',
+      },
+    };
+    mountWithTheme(
+      <ViewEditDashboard
+        location={TestStubs.location(location)}
+        organization={initialData.organization}
+        router={initialData.router}
+        params={{
+          orgId: initialData.organization.slug,
+          dashboardId: '1',
+        }}
+        route={{}}
+        routes={[]}
+        routeParams={{}}
+      >
+        {() => undefined}
+      </ViewEditDashboard>
+    );
+
+    expect(browserHistory.replace).toHaveBeenNthCalledWith(
+      1,
```

Comment:
Checked the docs real quick, indexing with `toHaveBeenNthCalledWith` starts from 1, is there a reason we chose this instead of `toHaveBeenCalledWith`?

Reply:
nope, no reason. Updated to `toHaveBeenCalledWith`!